File: solutions/2026-02-03/cpp/2026-02-03-trionic-array-i.cpp
```cpp
// Problem: Trionic Array I
// Link: https://leetcode.com/problems/trionic-array-i/
//
// Approach:
// The problem asks us to find if there exist two indices p and q such that 0 < p < q < n-1,
// and the array can be split into three segments with specific monotonicity properties:
// 1. nums[0...p] is strictly increasing.
// 2. nums[p...q] is strictly decreasing.
// 3. nums[q...n-1] is strictly increasing.
//
// We can iterate through all possible pairs of (p, q) that satisfy 0 < p < q < n-1.
// For each pair, we check if the three segments meet the required conditions.
//
// To check if a segment is strictly increasing, we iterate from the start of the segment
// to the end, ensuring each element is strictly greater than the previous one.
// To check if a segment is strictly decreasing, we iterate from the start of the segment
// to the end, ensuring each element is strictly less than the previous one.
//
// If we find any pair (p, q) that satisfies all conditions, we return true.
// If we exhaust all possible pairs without finding a valid one, we return false.
//
// Time Complexity: O(n^3)
// We have nested loops for p and q, which take O(n^2) time. Inside these loops,
// checking the three segments takes O(n) time in total. Thus, the overall complexity is O(n^3).
// Given the constraint n <= 100, n^3 is at most 1,000,000, which is acceptable.
//
// Space Complexity: O(1)
// We are only using a few variables to store indices and loop counters, so the space complexity is constant.

#include <vector>
#include <iostream>

class Solution {
public:
    bool isTrionic(std::vector<int>& nums) {
        int n = nums.size();

        // Iterate through all possible values of p.
        // p must be greater than 0 and less than n-2 (to allow for q and the last element).
        for (int p = 1; p < n - 2; ++p) {
            // Iterate through all possible values of q.
            // q must be greater than p and less than n-1 (to allow for the last element).
            for (int q = p + 1; q < n - 1; ++q) {
                // Assume the current (p, q) pair forms a trionic array and check.
                bool is_valid = true;

                // Check the first segment: nums[0...p] strictly increasing.
                // We start from index 1 to compare with the previous element.
                for (int i = 1; i <= p; ++i) {
                    if (nums[i] <= nums[i - 1]) {
                        is_valid = false;
                        break; // This segment is not strictly increasing.
                    }
                }

                // If the first segment is valid, check the second segment.
                if (is_valid) {
                    // Check the second segment: nums[p...q] strictly decreasing.
                    // We start from index p+1 to compare with the previous element.
                    for (int i = p + 1; i <= q; ++i) {
                        if (nums[i] >= nums[i - 1]) {
                            is_valid = false;
                            break; // This segment is not strictly decreasing.
                        }
                    }
                }

                // If the first two segments are valid, check the third segment.
                if (is_valid) {
                    // Check the third segment: nums[q...n-1] strictly increasing.
                    // We start from index q+1 to compare with the previous element.
                    for (int i = q + 1; i < n; ++i) {
                        if (nums[i] <= nums[i - 1]) {
                            is_valid = false;
                            break; // This segment is not strictly increasing.
                        }
                    }
                }

                // If all three segments are valid for this (p, q) pair, return true.
                if (is_valid) {
                    return true;
                }
            }
        }

        // If no such (p, q) pair was found after checking all possibilities, return false.
        return false;
    }
};
```

File: solutions/2026-02-03/java/2026-02-03-trionic-array-i.java
// Summary: Checks if an array can be divided into three segments: strictly increasing, strictly decreasing, and strictly increasing.
// Link: https://leetcode.com/problems/trionic-array-i/
// Approach:
// The problem requires us to find two indices p and q such that 0 < p < q < n-1, and the array segments nums[0...p], nums[p...q], and nums[q...n-1] satisfy the given increasing/decreasing conditions.
// We can iterate through all possible pairs of (p, q) that satisfy the index constraints.
// For each pair (p, q), we check if:
// 1. The subarray from index 0 to p is strictly increasing.
// 2. The subarray from index p to q is strictly decreasing.
// 3. The subarray from index q to n-1 is strictly increasing.
// If all three conditions are met for any pair (p, q), we return true.
// If no such pair is found after checking all possibilities, we return false.
//
// Time Complexity: O(n^3) - We have nested loops for p and q (O(n^2) pairs). For each pair, we iterate through the segments to check the increasing/decreasing property, which takes O(n) time in the worst case.
// Space Complexity: O(1) - We only use a few variables to keep track of indices and loop counters.
class Solution {
    public boolean isTrionic(int[] nums) {
        int n = nums.length;

        // Iterate through all possible values of p.
        // p must be greater than 0 and less than n-2 (to allow for q and the last element).
        for (int p = 1; p < n - 2; p++) {
            // Iterate through all possible values of q.
            // q must be greater than p and less than n-1 (to allow for the last increasing segment).
            for (int q = p + 1; q < n - 1; q++) {

                // Check if the first segment (nums[0...p]) is strictly increasing.
                boolean firstSegmentIncreasing = true;
                for (int i = 0; i < p; i++) {
                    if (nums[i] >= nums[i + 1]) {
                        firstSegmentIncreasing = false;
                        break;
                    }
                }

                // If the first segment is not increasing, continue to the next pair of (p, q).
                if (!firstSegmentIncreasing) {
                    continue;
                }

                // Check if the second segment (nums[p...q]) is strictly decreasing.
                boolean secondSegmentDecreasing = true;
                for (int i = p; i < q; i++) {
                    if (nums[i] <= nums[i + 1]) {
                        secondSegmentDecreasing = false;
                        break;
                    }
                }

                // If the second segment is not decreasing, continue to the next pair of (p, q).
                if (!secondSegmentDecreasing) {
                    continue;
                }

                // Check if the third segment (nums[q...n-1]) is strictly increasing.
                boolean thirdSegmentIncreasing = true;
                for (int i = q; i < n - 1; i++) {
                    if (nums[i] >= nums[i + 1]) {
                        thirdSegmentIncreasing = false;
                        break;
                    }
                }

                // If all three segments satisfy the conditions, return true.
                if (thirdSegmentIncreasing) {
                    return true;
                }
            }
        }

        // If no such pair of (p, q) is found after checking all possibilities, return false.
        return false;
    }
}
